http-common/src/connector.rs

```diff
@@ -80,8 +80,8 @@ impl Incoming {
                 let (unix_stream, _) = listener.accept().await?;
                 let ucred = unix_stream.peer_cred()?;
                 let user_state = user_state
-                    .lock()
-                    .await
+                    .get_mut()
```

Comment:
Oh, this already receives `&mut self`, so even a `std::sync::Mutex` is unnecessary. You can just have `user_state: BTreeMap` directly.

Reply:
Good catch. Fixed.